src/Components/MainTableView/UpdateTuple.tsx

```diff
@@ -0,0 +1,510 @@
+      );
+    }
+    else if (tableAttribute.attributeType === TableAttributeType.TIMESTAMP) {
+      return (
+        <div className="fieldUnit" key={JSON.stringify(tableAttribute)}>
+          {this.getAttributeLabelBlock(tableAttribute, 'time stamp')}
+          <div className="dateTimeFields">
+            <input type='date' disabled={tableAttribute.constructor === PrimaryTableAttribute ? true: false} defaultValue={defaultValue} id={tableAttribute.attributeName + '__date'} onChange={this.handleChange.bind(this, tableAttribute.attributeName + '__date')}></input>
+            <input type='time' disabled={tableAttribute.constructor === PrimaryTableAttribute ? true: false} step="1" defaultValue={defaultValue} id={tableAttribute.attributeName + '__time'} onChange={this.handleChange.bind(this, tableAttribute.attributeName + "__time")}></input>
+          </div>
+        </div>
+      );
+    }
+    else if (tableAttribute.attributeType === TableAttributeType.ENUM) {
+      return (
+        <div className="fieldUnit" key={JSON.stringify(tableAttribute)}>
+          {this.getAttributeLabelBlock(tableAttribute, 'enum')}
+          <select onChange={this.handleChange.bind(this, tableAttribute.attributeName)}> {
+            tableAttribute.enumOptions?.map((enumOptionString: string) => {
+              return(<option selected={this.state.tupleBuffer[tableAttribute.attributeName] === enumOptionString} key={enumOptionString} value={enumOptionString}>{enumOptionString}</option>);
+          })}
+          </select>
+        </div>
+      )
+    }
+
+    // Handle number return types
+    if (type === 'number') {
+      return (
+      <div className="fieldUnit" key={JSON.stringify(tableAttribute)}>
+        {this.getAttributeLabelBlock(tableAttribute, typeString)}
+        <input disabled={tableAttribute.constructor === PrimaryTableAttribute ? true: false} value={this.state.tupleBuffer[tableAttribute.attributeName]} type={type} min={min} max={max} defaultValue={defaultValue} id={tableAttribute.attributeName} onChange={this.handleChange.bind(this, tableAttribute.attributeName)}></input>
+      </div>
+      )
+    }
+
+    throw Error('Unsupported Type found for attribute: ' + tableAttribute.attributeName);
+  }
+
+  /**
+   * Helper function specifically dealing with primary attributes where it checks for autoIncrement, if that is true then return a disable Input
+   * @param primaryTableAttribute PrimaryTableAttribute attribute object to mainly check for autoIncrement
+   */
+  getPrimaryAttributeInputBlock(primaryTableAttribute: PrimaryTableAttribute) {
+    if (primaryTableAttribute.autoIncrement === true) {
+      return(
+        <div className="fieldUnit" key={JSON.stringify(primaryTableAttribute)}>
+          {this.getAttributeLabelBlock(primaryTableAttribute, 'Auto Increment')}
+          <input disabled></input>
+        </div>
+      )
+    }
+    else {
+      return this.getAttributeInputBlock(primaryTableAttribute);
+    }
+  }
```

Comment:
This needed to be extracted out, since not only does update needs this, but filter does. Please wait for the filter pull request, will fix it there.

Reply:
I also agree it would be best to make this a generic method to be consumed by `SELECT`, `INSERT`, `UPDATE` operations.

@Synicix If it is pretty straightforward for you to do this but difficult to explain or suggest here in the PR (e.g. too many lines to change), I'd propose you make a PR with the updates to support this to `mahos:updateAction` so she can review and merge directly into this PR. You just need to create a branch locally on your side from her branch. Let me know if you have trouble with this.

Reply:
Yeah it is pretty straight forward, but in the intrest of time I am planning to merge this then resolve it later with another pull request to refactor those problems.